packages/util-waiter/src/utils/validate.ts

```diff
@@ -1,13 +1,19 @@
-import { ResolvedWaiterOptions } from "../waiter";
+import { WaiterOptions } from "../waiter";
 
 /**
  * Validates that waiter options are passed correctly
  * @param options a waiter configuration object
  */
-export const validateWaiterOptions = <Client>(options: ResolvedWaiterOptions<Client>): void => {
+export const validateWaiterOptions = <Client>(options: WaiterOptions<Client>): void => {
   if (options.maxWaitTime < 1) {
-    throw `WaiterOptions.maxWaitTime must be greater than 0`;
+    throw `WaiterConfiguration.maxWaitTime must be greater than 0`;
```

Comment:
nit: Throw Error object so that it contains the stack trace and other debugging information.
```suggestion
    throw new Error(`WaiterConfiguration.maxWaitTime must be greater than 0`);
```

Ditto for other lines.

Reply:
great call. Updated everything to use this